includes/managers/icons.php

```diff
@@ -183,30 +195,129 @@ public static function get_icon_manager_tabs_config() {
 		return array_values( array_merge( $tabs, self::get_icon_manager_tabs() ) );
 	}
 
-	private static function render_svg_icon( $value ) {
+	/**
+	 * is_font_awesome_inline
+	 *
+	 * @return bool
+	 */
+	private static function is_font_icon_inline_svg() {
+		return Plugin::$instance->experiments->is_feature_active( 'e_font_icon_svg' );
+	}
+
+	/**
+	 * render_svg_symbols
+	 *
+	 */
+	public static function render_svg_symbols() {
+		$symbols = apply_filters( 'elementor/icons_manager/svg_symbols', [] );
+
+		if ( ! $symbols ) {
+			return;
+		}
+
+		$svg = '<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">';
+
+		foreach ( $symbols as $symbol_id => $symbol ) {
+			$svg .= '<symbol id="' . $symbol_id . '" viewBox="0 0 ' . esc_attr( $symbol['width'] ) . ' ' . esc_attr( $symbol['height'] ) . '">';
+			$svg .= '<path d="' . esc_attr( $symbol['path'] ) . '"></path>';
+			$svg .= '</symbol>';
+		}
+
+		$svg .= '</svg>';
+
+		// TODO: Replace with print_unescaped_string.
+		echo $svg;
+	}
+
+	public static function get_icon_svg_data( $icon ) {
+		$font_family_manager = Font_Icon_Svg_Manager::get_font_family_manager( $icon['font_family'] );
+
+		$config = $font_family_manager::get_config( $icon );
+
+		return self::$data_manager->get_asset_data( $config );
+	}
+
+	/**
+	 * Get font awesome svg.
+	 * @param $icon array [ 'value' => string, 'library' => string ]
+	 *
+	 * @return bool|mixed|string
+	 */
+	public static function get_font_icon_svg( $icon, $attributes = [] ) {
+		// Load the SVG from the database.
+		$icon_data = self::get_icon_svg_data( $icon );
+
+		if ( ! $icon_data['path'] ) {
+			return '';
+		}
+
+		// Add the icon data to the symbols array for later use in page rendering process.
+		add_filter( 'elementor/icons_manager/svg_symbols', function( $symbols ) use ( $icon_data ) {
```

Comment:
Instead of using a filter, you can just add the symbols to a static variable

Reply:
Done.